src/include/storage/compression/compression.h

```diff
@@ -105,37 +108,82 @@ enum class CompressionType : uint8_t {
     INTEGER_BITPACKING = 1,
     BOOLEAN_BITPACKING = 2,
     CONSTANT = 3,
+    FLOAT = 4,
+};
+
+// used only for compressing floats/doubles
+struct ALPMetadata {
+    ALPMetadata() = default;
+    explicit ALPMetadata(const alp::state& alpState, common::PhysicalTypeID physicalType);
+
+    uint8_t exp;
+    uint8_t fac;
+    uint32_t exceptionCount;
+    uint32_t exceptionCapacity;
 };
 
 // Data statistics used for determining how to handle compressed data
 struct CompressionMetadata {
+    struct InPlaceUpdateLocalState {
+        struct FloatState {
+            size_t newExceptionCount;
+        } floatState;
+    };
+
```

Comment:
added this to keep track of exceptions across multiple `canUpdateInPlace()` calls in case updates to a single chunk are split across multiple calls.

let me know if anyone has better ideas of how to do this